soc/arm/microchip_mec/common/soc_dma.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/sys_io.h>
+#include <sys/__assert.h>
+#include <soc.h>
+
+/* #define SOC_DMA_DEBUG_ENABLE 1 */
+
+int soc_dma_init(void)
+{
+	mec_pcr_periph_slp_ctrl(PCR_DMA, 0);
+
+	struct dma_regs *regs = (struct dma_regs *)(DMA_BASE);
+
+	regs->ACTRST = MCHP_DMAM_CTRL_SOFT_RESET | MCHP_DMAM_CTRL_ENABLE;
+
+	uint32_t cnt = 256U;
+
+	while (cnt--) {
+		if (regs->ACTRST == 0) {
+			regs->ACTRST = MCHP_DMAM_CTRL_ENABLE;
+			return 0;
+		}
+	}
+
+	return -ETIMEDOUT;
+}
+
+int soc_dma_chan_init(uint8_t chan)
+{
+	if (chan >= MCHP_DMA_CHANNELS) {
+		return -EINVAL;
+	}
+
+	struct dma_chan_regs *regs =
+		(struct dma_chan_regs *)DMA_CHAN_BASE(chan);
+
+	if (regs->ACTV) {
+		regs->CTRL |= MCHP_DMA_C_XFER_ABORT;
+
+		uint32_t cnt = 256U;
+
+		while (cnt--) {
+			if (!(regs->CTRL & MCHP_DMA_C_BUSY_STS)) {
+				break;
+			}
+		}
+
+		if (regs->CTRL & MCHP_DMA_C_BUSY_STS) {
+			return -ETIMEDOUT;
+		}
+
+		regs->ACTV = 0U;
+	}
+
+	regs->CTRL = 0U;
+	regs->MSTART = 0U;
+	regs->MEND = 0U;
+	regs->IEN = 0U;
+	regs->ISTS = 0xFFFFFFFFul;
```

Comment:
Can we please use lower-case hex everywhere?

Also, what are the U and ul for? They seem superfluous. Also should fix this everywhere

Reply:
According to MISRA-C 2012 the suffixes shall be `U` for unsigned literals and `UL` for unsigned long literals: https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html#coding-guidelines

Reply:
Oh dear I didn't know that. It seems silent on 0 though.

Also, what is unsigned about 0xffffffff? It is a bit field I think.

https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_07_02.c

Reply:
Some of the mixed use U and UL is due to my lack of knowledge: how does GCC determine the size of U, UL, and ULL?  Are U and UL the same size (32-bits) when GCC builds for 32-bit ARM Cortex-Mx?  We had issues ~5 years ago with a GCC based compiler truncating defines using U to 16 bits.  I have no problem using a uniform style if I know what the compiler will do.